app/src/main/java/org/mozilla/fenix/search/SearchInteractor.kt

```diff
@@ -25,6 +25,9 @@ class SearchInteractor(
     private val navController: NavController,
     private val store: SearchStore
 ) : AwesomeBarInteractor, ToolbarInteractor {
+
+    private var startedTyping = Pair(false, !store.state.showShortcutEnginePicker)
```

Comment:
Can you help me better understand what this `Pair` is doing? I wonder this could be represented better with a different data structure for more clarity on what the difference between `first` and `second` is.

Reply:
Originally, startedTyping just a simple boolean variable to check whether the user started typing (via onTextChanged). However, I noticed that onTextChanged is called when the SearchFragment is loaded, so I needed another variable to check this initial call.

Reply:
Instead of checking if it was the first call, perhaps we could just check `store.state.query != text`?